.github/workflows/pr_checks.yml

```
@@ -0,0 +1,245 @@
+name: PR Checks
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+    tags: '*'
+concurrency:
+  # Skip intermediate builds: all builds except for builds on the `master` branch
+  # Cancel intermediate builds: only pull request builds
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.ref != 'refs/heads/master' || github.run_number }}
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
+permissions:
+  contents: read
+jobs:
+  finalize-pr-checks:
+    if: always() # this line is important to keep the `finalize` job from being marked as skipped; do not change or delete this line
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    needs: [
+      checked-in-files,
+      build,
+      npm-run-test,
+      make-targets,
+      stalecheck-npm-install,
+    ]
```

The more canonical YAML way of doing this:
```suggestion
    needs:
      - checked-in-files
      - build
      - npm-run-test
      - make-targets
      - stalecheck-npm-install
```